core/src/main/java/org/apache/carbondata/core/scan/expression/conditional/FilterUtil.java

```diff
@@ -0,0 +1,133 @@
+          new LiteralExpression(value, DataTypes.BYTE));
+    } else {
+      throw new IllegalArgumentException("Unsupported data type: " + dataType);
+    }
+  }
+
+  public static Expression prepareOrExpression(List<Expression> expressions) {
+    if (expressions.size() < 2) {
+      throw new RuntimeException("Please input at least two expressions");
+    }
+    Expression expression = expressions.get(0);
+    for (int i = 1; i < expressions.size(); i++) {
+      expression = new OrExpression(expression, expressions.get(i));
+    }
+    return expression;
+  }
+
+  private static Expression prepareEqualToExpressionSet(String columnName, DataType dataType,
+      List<Object> values) {
+    Expression expression = null;
+    if (0 == values.size()) {
+      expression = prepareEqualToExpression(columnName, dataType.getName(), null);
+    } else {
+      expression = prepareEqualToExpression(columnName, dataType.getName(), values.get(0));
+    }
+    for (int i = 1; i < values.size(); i++) {
+      Expression expression2 = prepareEqualToExpression(columnName,
+          dataType.getName(), values.get(i));
+      expression = new OrExpression(expression, expression2);
+    }
+    return expression;
+  }
+
+  public static Expression prepareEqualToExpressionSet(String columnName, String dataType,
+      List<Object> values) {
+    if (DataTypes.STRING.getName().equalsIgnoreCase(dataType)) {
+      return prepareEqualToExpressionSet(columnName, DataTypes.STRING, values);
+    } else if (DataTypes.INT.getName().equalsIgnoreCase(dataType)) {
+      return prepareEqualToExpressionSet(columnName, DataTypes.INT, values);
+    } else if (DataTypes.DOUBLE.getName().equalsIgnoreCase(dataType)) {
+      return prepareEqualToExpressionSet(columnName, DataTypes.DOUBLE, values);
+    } else if (DataTypes.FLOAT.getName().equalsIgnoreCase(dataType)) {
+      return prepareEqualToExpressionSet(columnName, DataTypes.FLOAT, values);
+    } else if (DataTypes.SHORT.getName().equalsIgnoreCase(dataType)) {
+      return prepareEqualToExpressionSet(columnName, DataTypes.SHORT, values);
+    } else if (DataTypes.BINARY.getName().equalsIgnoreCase(dataType)) {
+      return prepareEqualToExpressionSet(columnName, DataTypes.BINARY, values);
+    } else if (DataTypes.DATE.getName().equalsIgnoreCase(dataType)) {
+      return prepareEqualToExpressionSet(columnName, DataTypes.DATE, values);
+    } else if (DataTypes.LONG.getName().equalsIgnoreCase(dataType)) {
+      return prepareEqualToExpressionSet(columnName, DataTypes.LONG, values);
+    } else if (DataTypes.TIMESTAMP.getName().equalsIgnoreCase(dataType)) {
+      return prepareEqualToExpressionSet(columnName, DataTypes.TIMESTAMP, values);
+    } else if (DataTypes.BYTE.getName().equalsIgnoreCase(dataType)) {
+      return prepareEqualToExpressionSet(columnName, DataTypes.BYTE, values);
+    } else {
+      throw new IllegalArgumentException("Unsupported data type: " + dataType);
+    }
```

Comment:
you can try to remove duplicate code by using this method `org.apache.carbondata.core.util.DataTypeUtil#valueOf(java.lang.String)`

Reply:
return value is different.

Reply:
what you do in this function is to get the datatype(`org.apache.carbondata.core.metadata.datatype.DataTypes` ) from string, that method can do that for you to turn this function into one line or eliminate it.